### PR TITLE
Fix g12 for single nozzle

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Fix G12 for single nozzle and mixing extruders.

Add soft endstop clamping

### CODE REVIEW

**Comment by Roxy-3D** [comment]
Is there a reason not to merge this?   

I'm not sure but this might fix the single tool wipe problem.

File: Marlin/src/libs/nozzle.cpp [hunk 1]
```diff
@@ -159,22 +159,49 @@ Nozzle nozzle;
...
+        LIMIT( start[arrPos].A, soft_endstop.min.A, soft_endstop.max.A); \
+        LIMIT(middle[arrPos].A, soft_endstop.min.A, soft_endstop.max.A); \
+        LIMIT(   end[arrPos].A, soft_endstop.min.A, soft_endstop.max.A); \
+      }while(0)
+
+      LIMIT_AXIS(x);
+      LIMIT_AXIS(y);
+      LIMIT_AXIS(z);
+
+      const float arc_part = radius / M_PI;
```
Comment (thinkyhead) [inline Marlin/src/libs/nozzle.cpp]: The radius divided by 3.14 is interesting…. It looks like the radius itself is what you probably want here instead. The circles will extend from the middle of some point out to the radius distance.

Comment (thinkyhead) [reply Marlin/src/libs/nozzle.cpp:176]: @InsanityAutomation — I can't merge this without some kind of answer.

Comment (thinkyhead) [reply Marlin/src/libs/nozzle.cpp]: I'm going to assume I am correct and replace `arc_part` below with `radius`.

Comment (InsanityAutomation) [reply Marlin/src/libs/nozzle.cpp]: Sorry for the delay, it's been very crazy around here lately dealing with some stuff. Yes you are correct.

**Pull request merged**